### PR TITLE
Osx toolchain update

### PR DESCRIPTION
**Pull request by theuni**
tl;dr: Update to a newer stable toolchain and SDK for OSX without giving up
any backwards compatibility. We can move to clang 3.5 as a next step which
allows use to use libc++ and the 10.10 sdk, but we'll need to find a build that
works in gitian/travis first.

Switch to a new, better maintained fork of cctools:
https://github.com/tpoechtrager/cctools-port

I've forked this and will be working on it some as well:
https://github.com/theuni/cctools-port

This brings in:
cctools v862
ld64: v241.9

It also fixes 64bit builds, so there's no longer any need to use a 32bit clang.
Since clang is no longer tied to an old/crusty 32bit build, clang has been
upgraded to 3.3. Unfortunately, there's a bug in 3.4 that breaks builds. 3.5
works fine, but there are no binary builds compatible with precise, which is
currently used for gitian and travis. We could always build our own if
necessary.

After updating to stable clang/linker/cctools, it's possible to use a more
recent SDK. The current SDK (10.7) through the most recent 10.10 have all been
built/tested successfully, both with and without 10.6 compatibility. However,
10.10 requires clang 3.5.

SDKs >= 10.9 use libc++ rather than libstdc++. This is verified working as well.

Sorry for the thought-dump, there are lots of pieces here. Despite that, this should be a pretty smooth upgrade. The messier parts (bumping the minimum runtime requirement, switching build SDKs, switching to libc++, rolling our own clang, etc) are left for later. All of those things have been tested and work fine, they just need to be discussed first.

### CODE REVIEW

**Comment by fanquake** [comment]
Nice. Tested. This fixes the Qt issue of building on 10.9 with the 10.10 SDK.

**Comment by laanwj** [comment]
Thanks for figuring this all out @theuni!
Is this intended for 0.10rc2 or just master?

**Comment by theuni** [comment]
@laanwj master only. I don't foresee any problems, but it'd be far too risky for 0.10 at the last minute.

**Comment by theuni** [comment]
First commit dropped, and rebased on top of #5586

**Comment by laanwj** [comment]
utACK
